### PR TITLE
Feature/tao 4972/new task queue

### PR DESCRIPTION
**Pull request by gyszucs**
Having AWS SQS implementation properly, we had to re-design the Task Queue from the ground.
I would suggest reading the [README](https://github.com/oat-sa/generis/blob/feature/TAO-4972/new-task-queue/common/oatbox/TaskQueue/README.md) about the Task Queue first.

### CODE REVIEW

File: common/oatbox/TaskQueue/AbstractTask.php [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+namespace oat\oatbox\TaskQueue;
+
+/**
+ * Class AbstractTask
+ *
+ * @author Gyula Szucs <gyula@taotesting.com>
+ */
+abstract class AbstractTask extends Message implements TaskInterface
+{
+    private $parameters = [];
```
Comment (hutnikau) [inline common/oatbox/TaskQueue/AbstractTask.php]: Should it be private? Classes which will extend `AbstractTask` will be not able to work with this property.

File: common/oatbox/TaskQueue/MessageBroker/AbstractMessageBroker.php [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+     * Denormalize the given message JSON.
+     *
+     * If the json is not valid, it deletes the message straight away without processing it.
+     *
+     * @param string $messageJSON
+     * @param string $receiptForDeletion An identification of the given message
+     * @param array  $logContext
+     * @return null|MessageInterface
+     */
+    protected function denormalizeMessage($messageJSON, $receiptForDeletion, array $logContext = [])
```
Comment (hutnikau) [inline common/oatbox/TaskQueue/MessageBroker/AbstractMessageBroker.php]: It seems more like unserialize (or decode) than denormalize

File: common/oatbox/TaskQueue/Message.php [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+     */
+    public function setCreatedAt(\DateTime $dateTime)
+    {
+        $this->setMetadata('_created_at_', $dateTime);
+
+        return $this;
+    }
+
+    /**
+     * @return \DateTime
```
Comment (hutnikau) [inline common/oatbox/TaskQueue/Message.php]: I guess after call of `denormalizeMessage` it will be array but not `\DateTime`

**Review by hutnikau — CHANGES_REQUESTED** [review]

**Comment by gyszucs** [comment]
Closing it because more work on it is required and will be split into smaller PRs.

**Pull request closed without merging**